src/ore/src/process.rs

```diff
@@ -45,3 +45,73 @@ macro_rules! halt {
             ::std::process::exit(2);
     }}
 }
+
+#[cfg(feature = "metrics")]
+pub mod rusage {
+    //! Report rusage metrics.
+
+    use crate::metrics::MetricsRegistry;
+    use prometheus::IntGauge;
+    use std::time::Duration;
+
+    macro_rules! metrics {
+        ($(($name:ident, $desc:expr)),*) => {
+            metrics! { @define $(($name, $desc)),*}
+        };
+        (@define $(($name:ident, $desc:expr)),*) => {
+            struct RuMetrics {
+                $($name: IntGauge,)*
+            }
+            impl RuMetrics {
+                fn new(registry: &MetricsRegistry) -> Self {
+                    Self {
+                        $($name: registry.register(crate::metric!(
+                            name: stringify!($name),
```

Comment:
I'm not quite sure about the names here. Should they have an `mz_` prefix at least?

FWIF, there is some official guidance about [Prometheus metrics naming conventions](https://prometheus.io/docs/practices/naming/). I think we're trying to follow that somewhat we our existing metrics, but we are not consistent. Maybe there are also Mz-specific guidelines somewhere, but I couldn't find any. Might be worth checking in with #eng-wg-internal-observability.

Reply:
Added a prefix of `mz_metrics_libc_`.